backend/entities/ProjectSnapshot.cs

```diff
@@ -23,22 +24,59 @@ public class ProjectSnapshot : BaseEntity
         /// </summary>
         public Project Project { get; set; }
 
+        /// <summary>
+        /// get/set - A name to identify the snapshot.
+        /// </summary>
+        /// <value></value>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// get/set - Whether this snapshot represents a final report (i.e. a report presented to the ADM).
+        /// </summary>
+        /// <value></value>
+        public bool IsFinal { get; set; }
+
+        /// <summary>
+        /// get/set - The date the snapshot was taken.
+        /// </summary>
+        /// <value></value>
+        public DateTime SnapshotOn { get; set; }
```

Comment:
Can we rename this to something like `SnapshotTimestamp`, `SnapshotOn` feels like its a boolean value